generator/src/dev-server.js

```diff
@@ -137,6 +137,21 @@ async function start(options) {
     base: options.base,
     ...viteConfig,
   });
+
+  let jsExists = false;
+  try {
+    fs.statSync("./port-data-source.js");
+    jsExists = true;
+  } catch (e) { }
+
+  let tsExists = false;
+  try {
+    fs.statSync("./port-data-source.ts");
+    tsExists = true;
+  } catch (e) { }
+
+  if (jsExists || tsExists) {
+    try {
```

Comment:
This is absolutely horrible, but I could neither find a way to set optional entry points in esbuild nor a nicer way of checking whether files exist.